numpy/lib/function_base.py

```diff
@@ -1320,6 +1320,477 @@ def gradient(f, *varargs, **kwargs):
     else:
         return outvals
 
+def divergence(v,*varargs,**kwargs):
```

Comment:
Some more general comments -- did not make up my mind about everything, so some of it is more for discussion:
  * I know varargs is copied from gradient, but I am frankly not a fan, I think I would prefer varargs to be a single sequence argument.
  * There seems to be quite a lot of duplication with gradient, maybe we can just call gradient (possible add some arguments to gradient, such as `axis=None` to pick which derivatives to calculate or some such; that would also in some sense allow the stacked matrix kind of logic)